modules/ocf_stats/files/stats/bin/close-old-sessions

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python3
+"""Closes old sessions that haven't updated in a while"""
+
+import sys
+
+from ocflib.lab.stats import get_connection
+
+# Timeout in minutes after which a host should be considered dead
+# (if we don't receive an update within X minutes, end any open sessions)
+HOST_TIMEOUT = 3
+
+
+def close_old_sessions(ocfstats_pass):
```

Comment:
Fair enough, that is pretty clean, mostly because it has a connection and the credentials already, so that's nice. I think we'd just want to keep extra stuff out of ocfweb if they don't relate to the web side of things, because otherwise we'd eventually get a monolithic web application that has a ton of dependencies and stuff that takes forever to work with. That's just my long-term worry about it anyway.

Reply:
feels like an abuse of periodic functions, that is really meant to be a caching mechanism...

Reply:
Yeah, that's also a worry, and if you want to change anything in it then it has to be fully deployed, which us more of a pain than just pushing to puppet.

Reply:
i think cron is the right way to go here unless somebody wants to add real batch support to ocfweb

Reply:
I think so too since it's unrelated to the other functions of ocfweb and doesn't even need a web server in the first place to run (unlike the desktop check-in feature), so it seems like cron would be the right place to put it.

Reply:
ok